### DISCUSSION TITLE
How can I isolate Jaeger web components?

### DISCUSSION THREAD

**Discussion by 3tilley** [jaegertracing/jaeger#3385] [Q&A]
Hello all,

I'll start by saying that I'm not a JS developer, so apologies if this is a nonsensical question.

I'd like to use my own trace source and web server (in another language) to show a HTML trace in the beautiful way that Jaeger does. We currently emit JSON from open telemetry for our services, but would like to display a limited set of spans as they are shown in the Jaeger UI. Ideally we would supply `GET /trace/<trace_id>` and then the UI would then render that on a static page.

Is it possible to isolate the front-end elements like this? Is this something that other people have done?

**Comment by joe-elliott** [comment]
It sounds like you will need to add a feature request for jaeger-ui or just write something custom to achieve what you want, but I can't say for sure. Can you clarify your needs?

You can configure the jaeger UI to work with any backend as long as it implements all of the Jaeger endpoints correctly. You can also [embed the UI](https://www.jaegertracing.io/docs/1.28/frontend-ui/#embedded-mode) but I'm not sure if that's what you're asking for either. Any details would help.
